opennms-base-assembly/src/main/filtered/bin/fix-karaf-setup.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/sh -
+set -e # exit when a command fails
+
+OPENNMS_HOME="${install.dir}"
+# The user that OpenNMS needs to run as.
+RUNAS="root"
+DATA_DIR="$OPENNMS_HOME/data"
+
+myuser="$(id -u -n)"
+if [ x"$myuser" = x"$RUNAS" ]; then
+	true # all is well
+else
+	echo "Error: you must run this script as $RUNAS, not '$myuser'" >&2
+	exit 4	# According to LSB: 4 - user had insufficient privileges
+fi
+
+echo "This script will try to fix karaf configuration problems by:"
+echo "- pruning $DATA_DIR directory. This is where the Karaf cache sits."
+echo "- restore all Karaf related configuration files to a pristine state"
+echo "It might be a good idea to take a backup of $OPENNMS_HOME/etc"
+
+# make sure user wants to proceed
+echo "Are you ready to continue? (y/n)"
+read -r answer
+echo
+if [ "$answer" != "y" ]
+then
+   echo "Ok, goodbye!"
+   exit 0
+fi
+
+# Prune data directory, exept for history.txt
+echo "Pruning directory $DATA_DIR"
+find "$DATA_DIR" -mindepth 1 -maxdepth 1 -not -name 'history.txt' -exec rm -r {} \;
+
+# Restore files
+PRISTINE_DIR="$OPENNMS_HOME/share/etc-pristine"
```

Comment:
Need to make sure this works with the Debian packages as well.

Reply:
@j-white how do I test that?

Reply:
I confirmed this works. For future reference, you can pull the .deb package from the Bamboo build i.e. https://bamboo.opennms.org/browse/OPENNMS-OPENNMS381-DEB-5/artifact, and then install that on a Debian-based system. In this case, I tested on Ubuntu 18.04 using `sudo docker run -ti ubuntu:18.04 /bin/bash`.

Reply:
ah ok, so you meant by that comment that the script runs on a Debian-based system? I guess we are good then since I develop on Ubuntu :-)